### DISCUSSION TITLE
Handling of Google Mail adresses

### DISCUSSION THREAD

**Discussion by dragoangel** [postalserver/postal#1519] [Feature suggestions]
Gmail has 2 points how one user can use different email in gmail:

- add dots at any part of username
- change domain from gmail.com to googlemail.com

This means that if we suppress email user@gmail.com we also need not send mail to user@googlemail.com as well as any combination of u.s.e.r@gmail.com, etc.

Before checking Google email over suppression list we can simplify their emails to no dots and gmail.com domain, and same when we save this email to suppression list. E.g. how this done on rspamd:
https://github.com/rspamd/rspamd/blob/a52bfa87d61294bb93df3b916f008c72ae23b9c9/lualib/lua_util.lua#L250-L274

**Comment by adamcooke** [comment]
I don't think there's any real harm in treating different combinations of gmail address separately for the purposes of suppression. 

It's quite unlikely for an actual user of a mail server to be sending mail to more than one variation of a Gmail address anyway.

**Reply by dragoangel** [reply]
1. this not hard to implement :)
2. well gmail users can do abuse in that way :), why not?
3. same stuff about tags can be done '+'

**Reply by adamcooke** [reply]
It's not terribly hard to implement, I agree. I just don't see a situation where it matters if `joesmith@gmail.com` receives an email when `joe.smith@gmail.com` is already on the suppression list. The alternative will quickly be added anyway if it isn't appropriate and I'd very doubtful if Google will penalise any server for doing this.
